integrations/api_activitypub_person_test.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package integrations
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	user_model "code.gitea.io/gitea/models/user"
+	"code.gitea.io/gitea/modules/activitypub"
+	"code.gitea.io/gitea/modules/setting"
+	"github.com/go-fed/activity/pub"
+	"github.com/go-fed/activity/streams"
+	"github.com/go-fed/activity/streams/vocab"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestActivityPubPerson(t *testing.T) {
+	onGiteaRun(t, func(*testing.T, *url.URL) {
+		setting.Federation.Enabled = true
+		setting.Database.LogSQL = true
+		defer func() {
+			setting.Federation.Enabled = false
+			setting.Database.LogSQL = false
+		}()
+
+		username := "user2"
+		req := NewRequestf(t, "GET", fmt.Sprintf("/api/v1/activitypub/user/%s", username))
+		resp := MakeRequest(t, req, http.StatusOK)
+		assert.Contains(t, resp.Body.String(), "@context")
+		var m map[string]interface{}
+		_ = json.Unmarshal(resp.Body.Bytes(), &m)
+
+		var person vocab.ActivityStreamsPerson
+		resolver, _ := streams.NewJSONResolver(func(c context.Context, p vocab.ActivityStreamsPerson) error {
+			person = p
+			return nil
+		})
+		ctx := context.Background()
+		err := resolver.Resolve(ctx, m)
+		assert.Equal(t, err, nil)
+		assert.Equal(t, "Person", person.GetTypeName())
+		assert.Equal(t, username, person.GetActivityStreamsName().Begin().GetXMLSchemaString())
+		keyID := person.GetJSONLDId().GetIRI().String()
+		assert.Regexp(t, fmt.Sprintf("activitypub/user/%s$", username), keyID)
+		assert.Regexp(t, fmt.Sprintf("activitypub/user/%s/outbox$", username), person.GetActivityStreamsOutbox().GetIRI().String())
+		assert.Regexp(t, fmt.Sprintf("activitypub/user/%s/inbox$", username), person.GetActivityStreamsInbox().GetIRI().String())
+
+		pkp := person.GetW3IDSecurityV1PublicKey()
```

Comment:
Seems like documentation says this can be `nil` https://pkg.go.dev/github.com/go-fed/activity/streams/impl/activitystreams/type_application#ActivityStreamsApplication.GetW3IDSecurityV1PublicKey

```suggestion
		pkp := person.GetW3IDSecurityV1PublicKey()
		assert.NotNil(t, pkp)
```